source/extensions/filters/network/rbac/rbac_filter.cc

```diff
@@ -63,9 +63,14 @@ RoleBasedAccessControlFilterConfig::RoleBasedAccessControlFilterConfig(
                                                   action_validation_visitor_)),
       shadow_engine_(Filters::Common::RBAC::createShadowEngine(
           proto_config, context, validation_visitor, action_validation_visitor_)),
-      enforcement_type_(proto_config.enforcement_type()) {}
+      enforcement_type_(proto_config.enforcement_type()),
+      delay_deny_ms_(PROTOBUF_GET_MS_OR_DEFAULT(proto_config, delay_deny, 0)) {}
```

Comment:
maybe set default value to -1?

I think 0 value is effectively "push to next dispatcher cycle" which is also reasonable

Reply:
does "push to next dispatcher cycle" give end user any benefit? If this is 0 or unset, we maintain the same behavior as today (close immediately) which doesn't enable the timer.